### PR TITLE
ci: Add `test-success` job

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

This allows not to specify particular jobs in matrix as required

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -296,6 +296,16 @@ jobs:
         uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d # v3.1.4
         if: always() && env.coverage == 'true'
```
Comment (viceice) [inline .github/workflows/build.yml:297]: we should change it too
```suggestion
         if: (success() || failure()) && env.coverage == 'true'
```

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: .github/workflows/build.yml (hunk 1 above)
Comment (zharinov) [reply .github/workflows/build.yml:297]: But if codecov servers are unavailabe for quite a while, we're in trouble

Comment (zharinov) [reply .github/workflows/build.yml:297]: Ah, I see, we're in trouble now :)

**Review by viceice — DISMISSED** [review]

**Pull request merged**